### PR TITLE
[Rule Tuning] Microsoft 365 Inbox Forwarding Rule Created

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Fix incorrect field pattern. 
Related to https://github.com/elastic/integrations/issues/2519

### CODE REVIEW

File: rules/integrations/o365/collection_microsoft_365_new_inbox_rule.toml [hunk 1]
```diff
@@ -43,9 +43,9 @@ query = '''
 event.dataset:o365.audit and event.provider:Exchange and
 event.category:web and event.action:"New-InboxRule" and
     (
-        o365audit.Parameters.ForwardTo:* or
-        o365audit.Parameters.ForwardAsAttachmentTo:* or
-        o365audit.Parameters.RedirectTo:*
+        o365.audit.Parameters.ForwardTo:* or
```
Comment (brokensound77) [inline rules/integrations/o365/collection_microsoft_365_new_inbox_rule.toml:46]: so did these change or were they always wrong?

Comment (w0rk3r) [reply rules/integrations/o365/collection_microsoft_365_new_inbox_rule.toml:46]: I've introduced them when I tuned the logic of this rule.
PR: https://github.com/elastic/detection-rules/pull/1683

**Review by imays11 — APPROVED** [review]
Checks:
- [ ] Date is adjusted
- [ ] all impacted field patterns changed correctly 

Looks Good

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Review by terrancedejesus — APPROVED** [review]
LGTM.

Checks:

- Checked for other rules affected
- non-ecs-schema.json changes appear correct
- adjustments to affected rule look correct

File: rules/integrations/o365/collection_microsoft_365_new_inbox_rule.toml (hunk 1 above)
Comment (brokensound77) [reply rules/integrations/o365/collection_microsoft_365_new_inbox_rule.toml:46]: oh, so the correct fields may exist in the parsed filebeat schema. If you remove those fields from the non-ecs schema, does this pass

Comment (w0rk3r) [reply rules/integrations/o365/collection_microsoft_365_new_inbox_rule.toml:46]: I'll try it

**Review by brokensound77 — APPROVED** [review]

**Pull request merged**